packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx

```diff
@@ -0,0 +1,73 @@
+import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
+
+@Component({
+    tag: 'va-segmented-progress-bar',
+    styleUrl: 'va-segmented-progress-bar.css',
+    shadow: true,
+})
+export class VaSegmentedProgressBar {
+    /**
+     * Whether or not an analytics event will be fired.
+     */
+    @Prop() enableAnalytics: boolean;
+
+    /**
+     * The current segment in progress
+     */
+    @Prop() current: number;
+
+    /**
+     * The total number of segments in the progress bar
+     */
+    @Prop() total: number;
+
+    /**
+    * An override for the default aria label.
+    */
+    @Prop() label: string = '';
+
+    @Event({
+        eventName: 'component-library-analytics',
+        composed: true,
+        bubbles: true,
+    })
+    componentLibraryAnalytics: EventEmitter;
+
+    componentDidRender() {
+        if (this.enableAnalytics) {
+            this.componentLibraryAnalytics.emit({
+                componentName: 'va-segmented-progress-bar',
+                action: 'change',
+                details: {
+                    current: this.current,
+                    total: this.total,
+                },
+            });
+        }
+    }
+
+    render() {
+        const label = this.label || `Step ${this.current} of ${this.total}`
```

Comment:
Alternatively, we could assign a default value here using something like:

```js
const {current, total, label = `Step ${current} of ${total}`} = this
```

Or at least, I think that's what the syntax would be.

Reply:
I can destructure the current and total off of this but doing the default label does not work. In the current syntax you are suggesting the label becomes an empty string and does not default to Step X of Y
<img width="231" alt="Screen Shot 2021-12-09 at 11 07 37 AM" src="https://user-images.githubusercontent.com/11822533/145432518-d80d33e6-39e4-40ec-a584-c49ef29186b3.png">

Reply:
@bkjohnson Ultimately I can make the code look like this:
// Label needs to use input or fallback and have instructions on correct one to use
// Needs to use initialized values as destructured values are causing data inconsistency
const label = this.label || Step ${this.current} of ${this.total}
const {current, total} = this

The other suggestions such as const {current, total, label = Step ${current} of ${total}} = this or 
const {current, total} = this and then const label = this.label || Step ${current} of ${total}
Result in the image I shared above with the label field being blank instead of showing Step X of Y